demisto_sdk/dev_tools/initiator.py

```diff
@@ -0,0 +1,254 @@
+import os
+import shutil
+import yaml
+import yamlordereddictloader
+
+from typing import Dict
+from distutils.dir_util import copy_tree
+from demisto_sdk.common.tools import print_error, print_color, LOG_COLORS
+from demisto_sdk.common.constants import INTEGRATIONS_DIR, SCRIPTS_DIR, INCIDENT_FIELDS_DIR, INCIDENT_TYPES_DIR, \
+    INDICATOR_FIELDS_DIR, PLAYBOOKS_DIR, LAYOUTS_DIR, TEST_PLAYBOOKS_DIR, CLASSIFIERS_DIR, CONNECTIONS_DIR, \
+    DASHBOARDS_DIR, MISC_DIR, REPORTS_DIR, WIDGETS_DIR
+
+
+class Initiator:
+    """Initiator creates a new pack/integration/script.
+
+       Attributes:
+           output_dir (str): The directory in which init will create the new pack/integration/script
+           name (str): The name for the new pack/integration/script
+           integration (bool): Indicates whether to create an integration.
+           script (bool): Indicates whether to create a script.
+           full_output_path (str): The full path to the newly created pack/integration/script
+    """
+
+    def __init__(self, output_dir: str, name: str, integration: bool = False, script: bool = False):
+        self.output_dir = output_dir if output_dir is not None else ''
+        self.name = name
+        self.is_integration = integration
+        self.is_script = script
+
+        self.full_output_path = ''
```

Comment:
i feel like the name and the dir name should be separated args due to the space issue

Reply:
I'll make sure a name has no spaces in it

Reply:
but it can have spaces, i feel like we should have two entities, one for file name and one for name in the yml

Reply:
done - added id parameter that can have spaces in it and is used for the id fields in the yml file of an integration or script.
keeping the "name" parameter for the use of directory name and file name which can not have spaces in it.

if no id given a prompt will appear asking for an id, or you can choose to use the current name as the id.